service/src/test/java/com/commercetools/util/HttpRequestUtilTimeoutTest.java

```diff
@@ -1,24 +1,112 @@
+     * </ol>
+     */
+    private static final String URL_PATTERN = "http://localhost:%d?delay=%d";
+
+    private static Logger LOGGER = LoggerFactory.getLogger(HttpRequestUtilTimeoutTest.class);
+
+    private int testRandomPort;
+
+    /**
+     * Mock http service instance, initiated separately for every test case.
+     *
+     * @see <a href="https://docs.oracle.com/javase/8/docs/jre/api/net/httpserver/spec/com/sun/net/httpserver/HttpServer.html">
+     * HttpServer</a>
+     */
+    private HttpServer server;
+
+
+    /**
+     * Create mock http service which responses after some <i>delay</i>, specified in the the GET query.
      */
-    private static final String SLOWWLY_URL_PATTERN = "http://slowwly.robertomurray.co.uk/delay/%d/url/http://google.com";
+    @Before
+    public void setUp() {
+        int retries = 3;
+        server = null;
+
+        // run the server on random port
+        do {
+            try {
+                // see https://en.wikipedia.org/wiki/Ephemeral_port#Range for the magic port number values range
+                testRandomPort = 49152 + (new Random(System.currentTimeMillis())).nextInt(65536 - 49152);
+                server = HttpServer.create(new InetSocketAddress(testRandomPort), 0);
+                break;
+            } catch (IOException e) {
+                retries--;
+                LOGGER.warn("Can't create mock http service on port {}, retry {} times...", testRandomPort, retries);
+            }
+        } while (retries > 0);
+
+        if (server == null) {
+            throw new IllegalStateException("Can't create mock http service after retries, last tried port number was " + testRandomPort);
+        }
+
+        // configure the server to "sleep" number of milliseconds specified in "?delay=" query argument
+        server.createContext("/", httpExchange -> {
+            List<NameValuePair> query = URLEncodedUtils.parse(httpExchange.getRequestURI(), StandardCharsets.UTF_8);
+
+            // read value from "?delay=" query argument
+            final long timeout = query.stream().filter(pair -> "delay".equals(pair.getName()))
+                    .map(NameValuePair::getValue)
+                    .mapToLong(Long::decode)
+                    .findFirst()
+                    .orElse(0);
+
+            LOGGER.info("Mock http server makes {} msec delay", timeout);
+
+            try {
+                sleep(timeout);
+            } catch (InterruptedException e) {
+                LOGGER.error("Mock http server sleeping error", e);
+            }
+
+            LOGGER.info("Mock http server continue response after {} msec ", timeout);
+
+            httpExchange.sendResponseHeaders(200, 0);
+            httpExchange.getResponseBody().close();
+        });
+
+        server.start();
+    }
+
+    @After
+    public void tearDown() {
+        if (server != null) {
+            server.removeContext("/");
+            server.stop(1);
```

Comment:
why 1 seconds as delay? and not 0? or more than 1?

Reply:
just got is from some example, maybe 1 is redundant:
baede087